controllers/mlmd.go

```diff
@@ -72,3 +78,47 @@ func (r *DSPAReconciler) ReconcileMLMD(dsp *dspav1alpha1.DataSciencePipelinesApp
 	log.Info("Finished applying MLMD Resources")
 	return nil
 }
+
+func (r *DSPAReconciler) GetEnvoyServiceHostname(ctx context.Context, dsp *dspav1alpha1.DataSciencePipelinesApplication) (string, error) {
+	service := &corev1.Service{}
+	namespacedNamed := types.NamespacedName{Name: "ds-pipeline-md-" + dsp.Name, Namespace: dsp.Namespace}
+	err := r.Get(ctx, namespacedNamed, service)
+	if err != nil {
+		return "", err
+	}
+
+	// Loop over all Service ports, if a secured port is found
+	// set port and scheme to its secured ones and skip the loop
+	serviceScheme := ""
+	servicePort := ""
+	for i := 0; i < len(service.Spec.Ports); i++ {
+		servicePort = fmt.Sprintf("%d", service.Spec.Ports[i].Port)
+		if servicePort == "8443" || servicePort == "443" {
+			// If a secured port is found, just set scheme to 'https://' and skip the loop
+			serviceScheme = "https://"
+			break
+		} else {
+			serviceScheme = "http://"
+		}
+	}
+
```

Comment:
this is identical to what you added in apiserver, so extract it into a utility function and call it in both places